addons/l10n_ph/utils.py

```diff
@@ -54,20 +56,20 @@ def _export_bir_2307(sheet_title, moves, file_format='xlsx'):
             'invoice_date': format_date(move.env, move.invoice_date, date_format="MM/dd/yyyy"),
             'vat': re.sub(r'-', '', partner.vat)[:9] if partner.vat else '',
             'branch_code': partner.branch_code or '000',
-            'company_name': partner.commercial_partner_id.name,
-            'first_name': partner.first_name or '',
-            'middle_name': partner.middle_name or '',
-            'last_name': partner.last_name or '',
-            'address': ', '.join([val for val in partner_address_info if val])
+            'company_name': partner.commercial_partner_id.name if partner.commercial_partner_id.company_type == 'company' else '',
+            'first_name': (partner.commercial_partner_id.first_name or '') if partner.company_type == 'person' else '',
```

Comment:
```suggestion
            'first_name': partner.company_type == 'person' and partner.commercial_partner_id.first_name or '',
```
For a more clear code you can assign the names by key because this requires to check company_type every time